src/components/Requirements/RequirementFulfillmentSlots.vue

```diff
@@ -118,10 +118,10 @@ export default defineComponent({
        * the nested requirement inside compound requirement the user chooses to display.
        */
       const matchedCourses = (
-        (this.requirementFulfillment.additionalRequirements || {})[
+        (this.requirementFulfillment.fulfillment.additionalRequirements || {})[
           this.compoundRequirementChoice
-        ] || this.requirementFulfillment
-      ).courses;
+        ] || this.requirementFulfillment.fulfillment
+      ).dangerousCourses;
```

Comment:
Here we are showing all the matching courses. We should use dangerous ones, since we want to show warning around those course cards. Using safe courses will eliminate the possibility of that.